api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -1215,4 +1220,59 @@ export class ScenariosController {
 
     return result.right;
   }
+
+  @ApiOperation({ description: 'Get PDF summary report for scenario' })
+  @ApiOkResponse()
+  @Header('content-type', 'application/pdf')
+  @Post('/:scenarioId/solutions/report')
+  async getSummaryReportForProject(
+    @Body() config: WebshotConfig,
+    @Param('scenarioId', ParseUUIDPipe) scenarioId: string,
+    @Res() res: Response,
+    @Req() req: RequestWithAuthenticatedUser,
+    @AppSessionToken() appSessionToken: string,
+  ): Promise<any> {
+    /**
+     * If a frontend app session token was provided via cookie, use this to let
+     * the webshot service authenticate to the app, otherwise fall back to
+     * looking for the relevant cookies in the body of the request.
+     *
+     * @todo Remove this once the new auth workflow via `Cookie` header is
+     * stable.
+     */
+    const configForWebshot = appSessionToken
+      ? {
+          ...config,
+          cookie: appSessionToken,
+        }
+      : config;
+    // @debt Refactor to use @nestjs/common's StreamableFile
+    // (https://docs.nestjs.com/techniques/streaming-files#streamable-file-class)
+    // after upgrading NestJS to v8.
```

Comment:
for sanity of our future selves reading this in months from now, this comment belongs just above the stream part below

Reply:
2208771b4c59372c2a5147b18156b7d3831414f6